src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/PaymentStepDetailsBlock/utils.ts

```diff
@@ -0,0 +1,50 @@
+import { BigNumber } from 'ethers';
+
+import { type ExpenditurePayoutWithSlotId } from '~types/expenditures.ts';
+import { type ExpenditurePayout } from '~types/graphql.ts';
+import { groupBy } from '~utils/lodash.ts';
+import { sortPayouts } from '~utils/sortPayouts.ts';
+
+export const getSummedTokens = (
+  data: ExpenditurePayoutWithSlotId[],
+  isPaid?: boolean,
+) => {
+  const groupedData = groupBy(data, 'tokenAddress');
```

Comment:
Thanks for bringing it here, now that it's next to the reduce it makes me wonder if we need to group before passing it further (instead of doing it inside reduce)?